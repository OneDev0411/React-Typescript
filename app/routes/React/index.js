// index.js
import React from 'react'
import { match, RoutingContext } from 'react-router'
import ReactDOMServer from 'react-dom/server'

// Store
import AppStore from '../../stores/AppStore'

// Config
import routes from './Config'

module.exports = (app, config) => {

  app.get('*',(req, res) => {

    match({ routes, location: req.url }, (error, redirectLocation, renderProps) => {
      
      let reactMarkup = ReactDOMServer.renderToStaticMarkup(<RoutingContext data={AppStore.data} {...renderProps} />)
      
      res.locals.reactMarkup = reactMarkup

      console.log(reactMarkup)

      if (error) {
      
        res.status(500).send(error.message)
      
      } else if (redirectLocation) {
        
        res.redirect(302, redirectLocation.pathname + redirectLocation.search)

      } else if (renderProps) {
        
        // Success!
        res.status(200).render('index.html')
      
      } else {

        res.status(404).render('index.html')
      
      }

    })
  })
  
}