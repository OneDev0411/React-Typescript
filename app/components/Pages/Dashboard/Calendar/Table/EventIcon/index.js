import React from 'react'

import { eventTypesIcons } from '../../../../../../views/utils/event-types-icons'
import { importantDatesIcons } from '../../../../../../views/utils/important-dates-icons'

const getIcon = Icon => (
  <Icon.icon
    style={{
      marginRight: '1rem',
      marginTop: '5px'
    }}
    fill={Icon.color}
  />
)

const getImportantDateIcon = event => {
  console.log('event', event)

  const { type_label: label, event_type: type } = event

  if (importantDatesIcons[label]) {
    return importantDatesIcons[label]
  }

  // Child or spouse birthdays
  // Todo: Refactor icons and related stuff
  if (type === 'birthday' || type === 'child_birthday') {
    return importantDatesIcons.Birthday
  }

  return eventTypesIcons['Task Critical']
}

const getCrmTaskIcon = type => {
  if (eventTypesIcons[type]) {
    return eventTypesIcons[type]
  }

  if (type === 'Message') {
    return eventTypesIcons.Mail
  }

  return eventTypesIcons['Task Critical']
}

const EventIcon = ({ event }) => {
  let icon

  switch (event.object_type) {
    case 'deal_context':
      icon = eventTypesIcons['Task Critical']
      break
    case 'crm_task':
      icon = getCrmTaskIcon(event.event_type)
      break

    case 'contact_attribute':
      icon = getImportantDateIcon(event)
      break

    default:
      icon = eventTypesIcons['Task Critical']
  }

  return getIcon(icon)
}

export default EventIcon
