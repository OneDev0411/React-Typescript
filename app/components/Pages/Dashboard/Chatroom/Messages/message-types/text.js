import React from 'react'
import moment from 'moment'

export default ({
  author,
  message,
  comment
}) => {

  return (
    <div>
      <div className="title">
        { author && author.abbreviated_display_name }
      </div>

      { comment }

      <p>
        { moment.unix(message.created_at).format('Y/M/D HH:mm:ss') }
      </p>
    </div>
  )
}
