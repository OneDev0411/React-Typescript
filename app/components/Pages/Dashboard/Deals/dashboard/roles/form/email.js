import React from 'react'
import cn from 'classnames'

export default ({ form, required, isInvalid, onChange }) => (
  <div className="input-container email">
    <label htmlFor="email" style={{ display: 'block', cursor: 'pointer' }}>
      Email {required && <sup>*</sup>}
    </label>
    <input
      id="email"
      name="email"
      type="email"
      required="required"
      placeholder="johnsmith@gmail.com"
      className={cn({ invalid: isInvalid })}
      value={form.email || ''}
      onChange={e => onChange(e.target.value)}
    />

    {isInvalid && <span className="error">Enter a valid email</span>}
  </div>
)
