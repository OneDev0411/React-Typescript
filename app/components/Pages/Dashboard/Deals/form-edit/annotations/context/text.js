import React from 'react'
import PDFJS from 'pdfjs-dist';

class Text extends React.Component {
  render() {
    const { rect, fontSize, font, bold, color, text } = this.props

    const style = {
      fontSize: (fontSize),
      fontFamily: font,
      color: color,
      fontWeight: bold ? 'bold' : 'normal',
      position: 'absolute',
      left: rect.left,
      top: rect.top,
      width: rect.width,
      height: rect.height,
    }

    return (
      <div
        type="text"
        style={style}
      >
        {text}
      </div>
    )
  }
}


export default Text
