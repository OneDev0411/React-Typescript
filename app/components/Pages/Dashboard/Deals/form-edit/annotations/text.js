import React from 'react'

import parseAppearanceString from './parseAppearanceString'
import cn from 'classnames'


class TextAnnotation extends React.Component {
  constructor(props) {
    super(props)
  }

  shouldComponentUpdate(props) {
    return props.value !== this.props.value
  }

  onValueUpdate(e) {
    const { value } = e.target
    this.props.onValueUpdate(value)
  }

  render() {
    const { annotation, value } = this.props
    const appearance = parseAppearanceString(annotation.defaultAppearance)

    const { rect } = annotation

    const style = {
      fontSize: (this.props.fontSize) + 'px',
      fontFamily: appearance.font,
      color: appearance.color,
      fontWeight: appearance.bold ? 'bold' : 'normal',
      position: 'absolute',
      left: rect[0],
      top: rect[1],
      width: Math.floor(rect[2] - rect[0]),
      height: Math.floor(rect[3] - rect[1]),
    }

    const { multiline } = rect

    return (
      <input
        id={ annotation.fieldName }
        type="text"
        key={ annotation.fieldName }
        style={style}
        onInput={ this.onValueUpdate.bind(this) }
        value={ value || '' }
      />
    )
  }
}


export default TextAnnotation
