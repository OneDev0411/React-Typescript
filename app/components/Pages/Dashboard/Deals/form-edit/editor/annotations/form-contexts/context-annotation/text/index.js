import React from 'react'

export default function Text(props) {
  const { rect, fontSize, font, bold, color, text } = props

  const style = {
    fontSize,
    fontFamily: font,
    color,
    fontWeight: bold ? 'bold' : 'normal',
    position: 'absolute',
    left: rect.left,
    top: rect.top,
    width: rect.width,
    height: rect.height
  }

  return (
    <div type="text" style={style}>
      {text}
    </div>
  )
}
