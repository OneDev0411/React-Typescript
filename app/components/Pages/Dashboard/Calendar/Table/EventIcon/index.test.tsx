import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

// Deal Context Icons
import IconTaskCritical from 'components/SvgIcons/TaskCritical/IconTaskCritical'

// Contact Dates (Important Dates) Icons
import IconBirthday from 'components/SvgIcons/Birthday/IconBirthday'

// CRM Tasks Icons
import IconCall from 'components/SvgIcons/CallOutline/IconCallOutline'
import IconText from 'components/SvgIcons/Text/IconText'
import IconChat from 'components/SvgIcons/Chat/IconChat'
import IconEmail from 'components/SvgIcons/EmailOutline/IconEmailOutline'
import IconMail from 'components/SvgIcons/Mail/IconMail'
import IconOther from 'components/SvgIcons/MenuRounded/IconMenuRounded'
import IconInPerson from 'components/SvgIcons/InPerson/IconInPerson'

// Deal Context Fixtures
import dealContext from 'fixtures/calendar/events/deal-context.json'

// CRM Tasks Fixtures
import call from 'fixtures/calendar/events/call.json'
import inPersonMeeting from 'fixtures/calendar/events/in-person-meeting.json'
import text from 'fixtures/calendar/events/text.json'
import chat from 'fixtures/calendar/events/chat.json'
import mail from 'fixtures/calendar/events/mail.json'
import email from 'fixtures/calendar/events/email.json'
import other from 'fixtures/calendar/events/other.json'

// Contact Dates (Important Dates) Fixtures
import birthday from 'fixtures/calendar/events/birthday.json'
import spouseBitrhday from 'fixtures/calendar/events/spouse-birthday.json'
import childBitrhday from 'fixtures/calendar/events/child-birthday.json'

import EventIcon from '.'

// TODO: Clean up icons and remove unused redundant ones
// and add all needed icons to this table
const EVENT_ICON_TABLE = [
  // Critical Dates
  {
    event: dealContext,
    icon: IconTaskCritical
  },
  // CRM Tasks
  {
    event: call,
    icon: IconCall
  },
  {
    event: inPersonMeeting,
    icon: IconInPerson
  },
  {
    event: text,
    icon: IconText
  },
  {
    event: chat,
    icon: IconChat
  },
  {
    event: mail,
    icon: IconMail
  },
  {
    event: email,
    icon: IconEmail
  },
  {
    event: other,
    icon: IconOther
  },
  // Contact Dates (Important Dates)
  {
    event: birthday,
    icon: IconBirthday
  },
  {
    event: spouseBitrhday,
    icon: IconBirthday
  },
  {
    event: childBitrhday,
    icon: IconBirthday
  }
]

describe('EventIcon component', () => {
  EVENT_ICON_TABLE.forEach(({ event, icon }) => {
    const wrapper = shallow(<EventIcon event={event} />)

    it(`renders ${event.type_label}`, () => {
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`renders proper icon for ${event.type_label}`, () => {
      expect(wrapper.find(icon)).toHaveLength(1)
    })
  })
})
