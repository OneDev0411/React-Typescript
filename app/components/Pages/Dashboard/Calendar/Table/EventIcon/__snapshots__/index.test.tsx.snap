// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventIcon component renders Birthday 1`] = `
<IconBirthday
  fill="#FF6F6F"
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M20 10.918a2.51 2.51 0 0 0-2.5-2.52h-2.918c.843-.319 1.744-.74 2.227-1.227.67-.674 1.07-1.547 1.07-2.333 0-.537-.179-1-.518-1.341-.831-.842-2.518-.583-3.645.556-.483.487-.9 1.395-1.216 2.245-.315-.85-.733-1.758-1.216-2.245-1.128-1.14-2.814-1.4-3.645-.556-.34.34-.517.804-.517 1.341-.002.786.399 1.66 1.069 2.333.483.488 1.384.908 2.227 1.226H7.5a2.51 2.51 0 0 0-2.5 2.52v2.521h1.25v5.041A2.51 2.51 0 0 0 8.75 21h7.5a2.51 2.51 0 0 0 2.5-2.52v-5.042H20v-2.52zm-5.4-5.974c.433-.438.981-.709 1.429-.709.136 0 .322.026.448.153s.151.315.151.451c.001.452-.268 1.005-.703 1.442-.414.417-1.52.868-2.508 1.192.32-.997.769-2.112 1.183-2.529zM8.372 4.84c0-.136.025-.325.15-.451.127-.127.313-.153.449-.153.448 0 .996.271 1.43.71.413.416.861 1.531 1.182 2.528-.989-.322-2.095-.775-2.508-1.192-.435-.437-.705-.99-.703-1.442zM6.25 10.918c0-.695.56-1.26 1.25-1.26h4.375v2.52H6.25v-1.26zm1.25 7.561v-5.04h4.375v6.3H8.75c-.69 0-1.25-.565-1.25-1.26zm10 0c0 .695-.56 1.26-1.25 1.26h-3.125v-6.3H17.5v5.04zm1.25-6.3h-5.625V9.657H17.5c.69 0 1.25.565 1.25 1.26v1.26z"
    fillRule="nonzero"
    key="key-0"
  />
</IconBirthday>
`;

exports[`EventIcon component renders Call 1`] = `
<IconCallOutline
  fill="#04C6D9"
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M19.96 16.181l-.018-.053c-.302-.693-2.079-2.204-2.15-2.275a8.69 8.69 0 0 0-.87-.676c-.179-.107-.392-.231-.658-.285-.622-.107-1.031.338-1.209.534-.018.018-.035.053-.07.07-.108.107-.232.267-.374.41a6.755 6.755 0 0 1-.587.64c-.285-.09-.853-.445-1.137-.676-1.173-.959-2.274-2.043-2.719-2.683l-.018-.019c-.48-.675-.64-1.012-.692-1.173a6.85 6.85 0 0 1 .64-.587c.16-.124.302-.248.409-.373l.07-.07c.213-.197.64-.587.533-1.21-.035-.266-.16-.48-.284-.657a8.61 8.61 0 0 0-.676-.87c-.07-.071-1.581-1.849-2.275-2.152l-.107-.036c-.426-.107-.817 0-1.19.302-.48.41-.96.889-1.44 1.493-.409.498-.818 1.066-1.013 1.742-.07.213-.32 1.102.178 2.329.089.302.515 1.723 1.777 3.27.782.959 2.826 3.358 5.012 4.922 1.51 1.067 2.736 1.511 2.967 1.582.587.248 1.102.32 1.511.32.444 0 .746-.089.853-.124.676-.196 1.226-.604 1.742-1.013.604-.48 1.084-.96 1.493-1.44.302-.374.409-.764.302-1.19v-.052zm-2.683 1.582c-.39.32-.818.622-1.244.746l-.035.018c-.018 0-.551.178-1.387-.16l-.07-.018c-.018 0-1.174-.373-2.595-1.386-2.098-1.51-4.16-3.963-4.728-4.674-1.19-1.458-1.529-2.773-1.529-2.773l-.035-.107c-.338-.818-.178-1.35-.178-1.368l.018-.054c.124-.444.444-.853.746-1.244.41-.515.8-.924 1.209-1.262.374.302 1.173 1.137 1.618 1.67.213.249.408.48.568.73v.017c-.018.019-.035.019-.035.036-.036.035-.071.053-.09.089a6.82 6.82 0 0 1-.337.302c-.586.498-.995.87-1.12 1.297-.177.587.214 1.315.943 2.364l.018.018c.622.907 1.937 2.133 2.968 2.986.07.054 1.528 1.263 2.434.978.426-.124.782-.533 1.298-1.12.124-.142.23-.267.302-.337l.089-.09.035-.035h.018c.248.16.498.356.729.569.533.462 1.386 1.262 1.67 1.618-.356.39-.765.781-1.28 1.19z"
    key="key-0"
  />
</IconCallOutline>
`;

exports[`EventIcon component renders Chat 1`] = `
<IconChat
  fill="#ff00bf"
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 100 100"
>
  <path
    d="M15.25 81.63a4 4 0 0 0 5.52 4.24l16.38-7A50.32 50.32 0 0 0 50 80.56c22.75 0 41.25-15 41.25-33.38S72.74 13.81 50 13.81 8.75 28.78 8.75 47.19a29.54 29.54 0 0 0 8.48 20.31zm1.5-34.44c0-14 14.92-25.37 33.25-25.37S83.25 33.2 83.25 47.19 68.33 72.56 50 72.56a42.07 42.07 0 0 1-12-1.71 4 4 0 0 0-2.7.15l-11.15 4.74 1.28-9.11a4 4 0 0 0-1.22-3.47c-4.88-4.58-7.46-10.1-7.46-15.97z"
    key="key-0"
  />
</IconChat>
`;

exports[`EventIcon component renders Child Birthday (Sam) 1`] = `
<IconBirthday
  fill="#FF6F6F"
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M20 10.918a2.51 2.51 0 0 0-2.5-2.52h-2.918c.843-.319 1.744-.74 2.227-1.227.67-.674 1.07-1.547 1.07-2.333 0-.537-.179-1-.518-1.341-.831-.842-2.518-.583-3.645.556-.483.487-.9 1.395-1.216 2.245-.315-.85-.733-1.758-1.216-2.245-1.128-1.14-2.814-1.4-3.645-.556-.34.34-.517.804-.517 1.341-.002.786.399 1.66 1.069 2.333.483.488 1.384.908 2.227 1.226H7.5a2.51 2.51 0 0 0-2.5 2.52v2.521h1.25v5.041A2.51 2.51 0 0 0 8.75 21h7.5a2.51 2.51 0 0 0 2.5-2.52v-5.042H20v-2.52zm-5.4-5.974c.433-.438.981-.709 1.429-.709.136 0 .322.026.448.153s.151.315.151.451c.001.452-.268 1.005-.703 1.442-.414.417-1.52.868-2.508 1.192.32-.997.769-2.112 1.183-2.529zM8.372 4.84c0-.136.025-.325.15-.451.127-.127.313-.153.449-.153.448 0 .996.271 1.43.71.413.416.861 1.531 1.182 2.528-.989-.322-2.095-.775-2.508-1.192-.435-.437-.705-.99-.703-1.442zM6.25 10.918c0-.695.56-1.26 1.25-1.26h4.375v2.52H6.25v-1.26zm1.25 7.561v-5.04h4.375v6.3H8.75c-.69 0-1.25-.565-1.25-1.26zm10 0c0 .695-.56 1.26-1.25 1.26h-3.125v-6.3H17.5v5.04zm1.25-6.3h-5.625V9.657H17.5c.69 0 1.25.565 1.25 1.26v1.26z"
    fillRule="nonzero"
    key="key-0"
  />
</IconBirthday>
`;

exports[`EventIcon component renders Email 1`] = `
<IconEmailOutline
  fill="#8F6CF0"
  fillColor={null}
  fillColorRule="&&& path, &&& use, &&& g"
  size={null}
  sizes={Object {}}
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M5 5.25h14A2.75 2.75 0 0 1 21.75 8v8A2.75 2.75 0 0 1 19 18.75H5A2.75 2.75 0 0 1 2.25 16V8A2.75 2.75 0 0 1 5 5.25zm0 1.5c-.69 0-1.25.56-1.25 1.25v8c0 .69.56 1.25 1.25 1.25h14c.69 0 1.25-.56 1.25-1.25V8c0-.69-.56-1.25-1.25-1.25H5z"
    key="key-0"
  />
  <path
    d="M12 12.003L4.494 5.436l-.988 1.128L12 13.997l8.494-7.433-.988-1.128z"
    key="key-1"
  />
</IconEmailOutline>
`;

exports[`EventIcon component renders Executed Date 1`] = `
<IconTaskCritical
  fill="#F5A623"
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M18.3 3H5.7C4.17 3 3 4.17 3 5.7v12.6C3 19.83 4.17 21 5.7 21h12.6c1.53 0 2.7-1.17 2.7-2.7V5.7C21 4.17 19.83 3 18.3 3zM5.7 4.8h.9v1.8h1.8V4.8h7.2v1.8h1.8V4.8h.9c.54 0 .9.36.9.9v1.8H4.8V5.7c0-.54.36-.9.9-.9zm12.6 14.4H5.7c-.54 0-.9-.36-.9-.9v-9h14.4v9c0 .54-.36.9-.9.9zm-7.2-8.1v3.6h1.8v-3.6h-1.8zm0 4.5v1.8h1.8v-1.8h-1.8z"
    fillRule="nonzero"
    key="key-0"
  />
</IconTaskCritical>
`;

exports[`EventIcon component renders In-Person Meeting 1`] = `
<IconInPerson
  fill="#F7A700"
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M12.744 14.474l.17.053c2.742.867 4.215 1.515 4.852 2.317.403.509.534 1.097.36 1.648-.194.621-.742 1.075-1.49 1.246-1.523.352-3.33.465-6.535.465-3.21 0-5.013-.113-6.536-.465-.747-.171-1.296-.625-1.49-1.246-.173-.552-.042-1.14.36-1.647.637-.803 2.11-1.452 4.852-2.318l.17-.053a5.197 5.197 0 0 0-.306-.56l-.064-.106c-.374-.614-.471-.837-.583-1.442a1.36 1.36 0 0 0-.23-.466c-.41-.455-.704-1.783-.487-2.484.01-.036.034-.094.065-.162a2.632 2.632 0 0 0 .057-.13 1.46 1.46 0 0 0 .03-.082.448.448 0 0 0 .027-.11c-.052-.828.024-1.6.228-2.295.3-1.026 1.16-1.946 2.198-2.355C8.948 4.062 9.422 4 10.1 4c.68 0 1.152.063 1.709.282 1.037.409 1.897 1.33 2.198 2.355.204.693.28 1.465.227 2.295 0 .012.008.054.027.11a1.468 1.468 0 0 0 .03.082 2.738 2.738 0 0 0 .057.13c.031.068.055.126.066.163.216.7-.077 2.028-.488 2.482a1.363 1.363 0 0 0-.23.466c-.077.42-.166.694-.313.978a7.05 7.05 0 0 1-.27.466l-.064.106c-.144.235-.243.42-.305.559zm4.084 3.563c.017-.054-.022-.154-.11-.266-.44-.554-1.919-1.142-4.195-1.86-.436-.138-.54-.173-.632-.216a1.009 1.009 0 0 1-.548-.655c-.146-.526.052-1.078.556-1.905l.064-.107c.285-.467.319-.546.392-.939.068-.373.313-.888.564-1.164.103-.153.25-.842.21-1.018-.192-.417-.282-.711-.259-1.073.042-.652-.016-1.248-.17-1.77-.174-.597-.728-1.178-1.37-1.43-.377-.15-.703-.191-1.23-.191s-.852.041-1.23.19c-.641.253-1.194.833-1.37 1.431-.153.523-.21 1.119-.17 1.77.018.271-.039.541-.148.82a3.784 3.784 0 0 1-.1.231c-.051.202.1.89.195 1.033.258.284.5.8.57 1.17.072.395.106.473.39.94l.065.105c.504.826.703 1.38.557 1.906-.082.296-.277.53-.55.657-.091.042-.198.078-.63.215-2.276.717-3.756 1.306-4.195 1.86-.089.112-.128.212-.111.268.03.099.197.223.484.29 1.409.325 3.136.431 6.243.431s4.834-.106 6.244-.432c.287-.066.453-.19.484-.29zm4.744-2.206c.338.427.452.944.304 1.416-.167.53-.627.916-1.251 1.06-.39.09-.824.16-1.323.214-.04.004-.04.004-.074.005-.353 0-.646-.284-.68-.65-.037-.394.232-.75.61-.79.454-.05.838-.112 1.175-.19a.647.647 0 0 0 .24-.099c-.014.017-.028-.012-.05-.038a1.387 1.387 0 0 0-.359-.279c-.29-.173-.706-.36-1.218-.555a25.708 25.708 0 0 0-1.515-.519c-.352-.11-.423-.135-.501-.17a.936.936 0 0 1-.507-.607c-.127-.457.032-.915.439-1.585l.05-.082c.207-.339.223-.376.274-.65.055-.304.243-.71.441-.932.07-.124.162-.552.152-.622-.152-.334-.225-.584-.206-.888a3.707 3.707 0 0 0-.12-1.277c-.118-.402-.498-.797-.937-.97-.264-.104-.494-.133-.873-.133-.18 0-.315.006-.448.023-.38.051-.72-.235-.765-.63-.045-.39.217-.752.593-.802.199-.026.384-.035.62-.035.536 0 .91.05 1.352.224.843.332 1.518 1.056 1.764 1.895.16.547.22 1.153.18 1.788 0 0 .018.06.068.169.027.058.048.111.06.148.183.592-.06 1.668-.409 2.053a.928.928 0 0 0-.13.264c-.09.485-.173.676-.466 1.156l-.032.053a5.804 5.804 0 0 0-.144.247c2.064.655 3.183 1.155 3.686 1.788z"
    key="key-0"
  />
</IconInPerson>
`;

exports[`EventIcon component renders Mail 1`] = `
<IconMail
  fill="#7ED321"
  fillColor={null}
  fillColorRule="&&& path, &&& use, &&& g"
  size={null}
  sizes={Object {}}
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M3.5 4.25h17a2.25 2.25 0 0 1 2.25 2.25v11a2.25 2.25 0 0 1-2.25 2.25h-17a2.25 2.25 0 0 1-2.25-2.25v-11A2.25 2.25 0 0 1 3.5 4.25zm0 1.5a.75.75 0 0 0-.75.75v11c0 .414.336.75.75.75h17a.75.75 0 0 0 .75-.75v-11a.75.75 0 0 0-.75-.75h-17z"
    key="key-0"
  />
  <path
    d="M5.5 13.25h3.75v-1.5H5.5zm0 3h8.25v-1.5H5.5zm10-9.5h3a1.5 1.5 0 0 1 1.5 1.5v3a1.5 1.5 0 0 1-1.5 1.5h-3a1.5 1.5 0 0 1-1.5-1.5v-3a1.5 1.5 0 0 1 1.5-1.5zm0 1.5v3h3v-3h-3z"
    key="key-1"
  />
</IconMail>
`;

exports[`EventIcon component renders Other 1`] = `
<IconMenuRounded
  fill="#9013FE"
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M20 12a8 8 0 1 0-8 8 8 8 0 0 0 8-8zM5.6 12a6.4 6.4 0 1 1 6.4 6.4A6.4 6.4 0 0 1 5.6 12zm1.3 0a1.288 1.288 0 1 0 2.576 0A1.288 1.288 0 0 0 6.9 12zm3.812 0a1.288 1.288 0 1 0 2.576 0 1.288 1.288 0 0 0-2.576 0zm3.812 0a1.288 1.288 0 1 0 2.576 0 1.288 1.288 0 0 0-2.576 0z"
    key="key-0"
  />
</IconMenuRounded>
`;

exports[`EventIcon component renders Spouse Birthday 1`] = `
<IconBirthday
  fill="#FF6F6F"
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M20 10.918a2.51 2.51 0 0 0-2.5-2.52h-2.918c.843-.319 1.744-.74 2.227-1.227.67-.674 1.07-1.547 1.07-2.333 0-.537-.179-1-.518-1.341-.831-.842-2.518-.583-3.645.556-.483.487-.9 1.395-1.216 2.245-.315-.85-.733-1.758-1.216-2.245-1.128-1.14-2.814-1.4-3.645-.556-.34.34-.517.804-.517 1.341-.002.786.399 1.66 1.069 2.333.483.488 1.384.908 2.227 1.226H7.5a2.51 2.51 0 0 0-2.5 2.52v2.521h1.25v5.041A2.51 2.51 0 0 0 8.75 21h7.5a2.51 2.51 0 0 0 2.5-2.52v-5.042H20v-2.52zm-5.4-5.974c.433-.438.981-.709 1.429-.709.136 0 .322.026.448.153s.151.315.151.451c.001.452-.268 1.005-.703 1.442-.414.417-1.52.868-2.508 1.192.32-.997.769-2.112 1.183-2.529zM8.372 4.84c0-.136.025-.325.15-.451.127-.127.313-.153.449-.153.448 0 .996.271 1.43.71.413.416.861 1.531 1.182 2.528-.989-.322-2.095-.775-2.508-1.192-.435-.437-.705-.99-.703-1.442zM6.25 10.918c0-.695.56-1.26 1.25-1.26h4.375v2.52H6.25v-1.26zm1.25 7.561v-5.04h4.375v6.3H8.75c-.69 0-1.25-.565-1.25-1.26zm10 0c0 .695-.56 1.26-1.25 1.26h-3.125v-6.3H17.5v5.04zm1.25-6.3h-5.625V9.657H17.5c.69 0 1.25.565 1.25 1.26v1.26z"
    fillRule="nonzero"
    key="key-0"
  />
</IconBirthday>
`;

exports[`EventIcon component renders Text 1`] = `
<IconText
  fill="#000"
  style={
    Object {
      "marginRight": "1rem",
      "marginTop": "5px",
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M16 4H8c-1.105 0-2 .796-2 1.778v12.444C6 19.204 6.895 20 8 20h8c1.105 0 2-.796 2-1.778V5.778C18 4.796 17.105 4 16 4zm1 14.222c0 .49-.448.888-1 .89H8c-.552-.001-1-.4-1-.89V5.778c0-.49.448-.889 1-.89h1c0 .492.448.89 1 .89h4c.552 0 1-.398 1-.89h1c.552.001 1 .4 1 .89v12.444zm-4-.889c0 .491-.448.89-1 .89s-1-.399-1-.89c0-.49.448-.889 1-.889.265 0 .52.094.707.26a.841.841 0 0 1 .293.63z"
    key="key-0"
  />
</IconText>
`;
